craft_parts/parts.py

```diff
@@ -205,8 +206,16 @@ def part_build_dir(self) -> Path:
 
     @property
     def part_build_subdir(self) -> Path:
-        """Return the subdirectory in build containing the source subtree (if any)."""
-        if self.spec.source_subdir:
+        """Return the subdirectory in build containing the source subtree (if any).
+
+        Parts that have a source subdirectory and do not support out-of-source builds
+        will have a build subdirectory.
+        """
+        if (
```

Comment:
This was the missing check from [snapcraft_legacy](https://github.com/snapcore/snapcraft/blob/e03a971b3c6bff8c1caa4f33502cfb6a8ef401eb/snapcraft_legacy/internal/pluginhandler/__init__.py#L106).